libs/components/datetime/src/lib/modules/date-range-picker/date-range-picker.component.html

```diff
@@ -11,115 +11,107 @@
     }"
   >
     <sky-input-box
-      [disabled]="disabled"
+      [hasErrors]="hasErrors"
       [helpPopoverContent]="helpPopoverContent"
       [helpPopoverTitle]="helpPopoverTitle"
+      [hintText]="hintText"
       [labelText]="
         label || ('skyux_date_range_picker_default_label' | skyLibResources)
       "
-      [hintText]="hintText"
     >
       <select
-        class="sky-form-control"
         formControlName="calculatorId"
-        [attr.id]="dateRangePickerId + '-select-calculator'"
-        (blur)="onFieldBlur()"
+        [required]="isRequired()"
+        (blur)="onBlur()"
       >
         <option
           *ngFor="let calculator of calculators"
           [value]="calculator.calculatorId"
         >
-          {{ calculator.shortDescription }}
+          {{
+            calculator._shortDescriptionResourceKey
+              ? (calculator._shortDescriptionResourceKey | skyLibResources)
+              : calculator.shortDescription
+          }}
```

Comment:
Instead of waiting for the `SkyDateRangeService` to load all the resource strings asynchronously (and loading an asynchronous task in `ngOnInit`), I'm passing the resources string to the `skyLibResources` pipe in the template.